Tests/SentryProfilerTests/SentryAppLaunchProfilingTests.swift

```diff
@@ -0,0 +1,190 @@
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        sentry_manageTraceProfilerOnStartSDK(options, TestHub(client: nil, andScope: nil))
+        XCTAssert(appLaunchProfileConfigFileExists())
+        options.profilesSampleRate = 0
+        sentry_manageTraceProfilerOnStartSDK(options, TestHub(client: nil, andScope: nil))
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        // ensure we get another config written, to test removal again
+        options.profilesSampleRate = 0.567
+        sentry_manageTraceProfilerOnStartSDK(options, TestHub(client: nil, andScope: nil))
+        XCTAssert(appLaunchProfileConfigFileExists())
+        options.tracesSampleRate = 0
+        sentry_manageTraceProfilerOnStartSDK(options, TestHub(client: nil, andScope: nil))
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+    }
+
+    // test continuous launch profiling configuration
+    func testContinuousLaunchProfileConfiguration() throws {
+        let options = Options()
+        options.enableContinuousProfiling = true
+        
+        // sample rates are not considered for continuous profiling
+        options.profilesSampleRate = 0
+        options.tracesSampleRate = 0
+        
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        sentry_manageTraceProfilerOnStartSDK(options, TestHub(client: nil, andScope: nil))
+        XCTAssert(appLaunchProfileConfigFileExists())
+        let dict = try XCTUnwrap(appLaunchProfileConfiguration())
+        XCTAssertEqual(dict[kSentryLaunchProfileConfigKeyContinuousProfiling], true)
+
+        _sentry_nondeduplicated_startLaunchProfile()
+        XCTAssert(SentryContinuousProfiler.isCurrentlyProfiling())
+    }
+    
+    func testTraceProfilerStartsWhenBothSampleRatesAreSet() {
+        let options = Options()
+        options.enableAppLaunchProfiling = true
+        options.profilesSampleRate = 0.567
+        options.tracesSampleRate = 0.789
+        XCTAssertFalse(appLaunchProfileConfigFileExists())
+        sentry_manageTraceProfilerOnStartSDK(options, TestHub(client: nil, andScope: nil))
+        XCTAssertTrue(appLaunchProfileConfigFileExists())
+        _sentry_nondeduplicated_startLaunchProfile()
+        XCTAssert(SentryLegacyProfiler.isCurrentlyProfiling())
+    }
+ 
+    /**
+     * Test how combinations of the following options interact to ultimately decide whether or not to start the profiler on the next app launch..
+     * - `enableLaunchProfiling`
+     * - `enableTracing`
+     * - `enableContinuousProfiling` (always profiles regardless of sample rate or trace options)
+     * - `tracesSampleRate`
+     * - `profilesSampleRate`
+     * - `profilesSampler`
+     */
+    func testShouldProfileLaunchBasedOnOptionsCombinations() {
+        for testCase: (enableAppLaunchProfiling: Bool, enableTracing: Bool, enableContinuousProfiling: Bool, tracesSampleRate: Int, profilesSampleRate: Int, profilesSamplerReturnValue: Int, shouldProfileLaunch: Bool) in [
```

Comment:
This looks nuts, but we now have a ton of combinations that can affect whether or not a launch profile starts. I originally tried writing this with Quick BDD style tests, but the code became unreadable (i've left the intermediate commits, if you're curious). I figured this is better because it explicitly lists all the options that lead to a launch profile starting. This PR has a sizable diff, but this file alone is half of it.

Reply:
I haven't heard that term before, thanks for the link. I'm not sure this is directly applicable in the way you think though. We don't have just one partition, but multiple, so we need to test all permutations. Just because one test case with `enableContinuousProfiling` set to true "covers" the lines of code in the implementation, it doesn't necessarily cover all the paths that traverse those lines of code. It's possible that in the future, someone puts an early return before those lines of code, giving an invalid result, due to one of the other options. Testing all the combinations protects against that. Am I missing something with the technique you mention?

Reply:
Also maybe that's what I was trying to do with the nested BDD stuff, if you look at those intermediate commits, but it got confusing to know what had been set for a given final test result. I found it better to just have each combination explicitly listed.

Reply:
I'm fine with keeping it as it is. We can discuss strategies to reduce the amount of test cases another time. We only need to change the tests when the requirements change. I'm OK with how it is right now.